ext/datadog_profiling_native_extension/stack_recorder.c

```diff
@@ -570,6 +614,8 @@ void record_sample(VALUE recorder_instance, ddog_prof_Slice_Location locations,
     labels.end_timestamp_ns
   );
 
+  active_slot.stats->recorded_samples++;
```

Comment:
I thought about this. I think at this point and paired with the rest of the internal info we have enough info to breakdown and thus can skip the extra complexity of tracking multiple types of samples:
* Heap samples written: `recorder_stats.heap_recorder_snapshot.last_update_objects_alive`.
* Allocation samples written: `worker_stats.allocation_sampled` 
* CPU/Wall samples: `profile_stats.recorded_samples - (last_update_objects_alive + allocation_sampled)`
And we can revisit in the future?

Reply:
I guess if we have the others is this one useful at all? :package: 

I'll leave the decision to you ;)

Reply:
IMO it is since it captures total libdatadog Adds for this profile and without it you can't really arrive at the `CPU/Wall samples` unless we tracked it in a separate metric.

Reply:
Ah. We do actually have this data I think; it's the sum of `CPU_SAMPLES_VALUE` in a profile (`cpu_or_wall_samples` counter).

But to be honest that gets reported in the pprof and we never did anything with it, so I'm not overly attached to it either. (And with timeline it's less useful, since you can recover this information by counting anyway).